### ISSUE TITLE
Provide t.n.srvconnect with full tests and do a general clean up.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2800]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2800|
|Type|task|
|Created|2007-08-24 08:42:07Z|

With the emergence of tests for `twisted.names.srvconnect`, it appears this module can use a lot of love in terms of clean up and more tests.

<details><summary>Searchable metadata</summary>

```
trac-id__2800 2800
type__task task
reporter__ralphm ralphm
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1187944927000000 1187944927000000
changetime__1465913311552501 1465913311552501
version__None None
owner__Ralph_Meijer__ralphm_____ Ralph Meijer <ralphm@...>
cc__ralphm cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm set owner to @ralphm<br>@ralphm set status to `closed`|
|-|-|

In changeset 5aee182293607ad4b806d81c26dbbd1f6e1f4624
```
#!CommitTicketReference repository="" revision="5aee182293607ad4b806d81c26dbbd1f6e1f4624"
Merge py3-srvconnect-8262: Port twisted.names.srvconnect to Python 3.

Author: ralphm
Reviewer: adiroiban, hawkowl, glyph
Fixes: #8262, #5124, #2800

This also cleans up the use of a cmp function for sorting SRV records,
makes the `pickServer` implementation in line with the algorithm
specified in the RFC and adds tests around that to match.
```
